catalog/core/catalog-core-downloadaction/src/test/java/org/codice/ddf/catalog/download/action/ResourceDownloadActionProviderTest.java

```diff
@@ -0,0 +1,250 @@
+        setupMockBasicMetacard(REMOTE_SITE_NAME, CONTENT_RESOURCE_URI, DEFAULT_METACARD_ID);
+        setupMockResourceCacheServiceMBeanProxy(false);
+        assertThat(actionProvider.canHandleMetacard(mockMetacard), is(true));
+    }
+
+    /**
+     * Metacard source id is local, Metacard contains remote resource uri, resource is cached; no
+     * action returned
+     */
+    @Test
+    public void canHandleLocalSiteRemoteUriResourceCached() throws Exception {
+        setupMockBasicMetacard(LOCAL_SITE_NAME, REMOTE_RESOURCE_URI, DEFAULT_METACARD_ID);
+        setupMockResourceCacheServiceMBeanProxy(true);
+        assertThat(actionProvider.canHandleMetacard(mockMetacard), is(false));
+    }
+
+    /**
+     * Metacard source id is local, Metacard contains remote resource uri, resource is not cached;
+     * action returned
+     */
+    @Test
+    public void canHandleLocalSiteRemoteUriResourceNotCached() throws Exception {
+        setupMockBasicMetacard(LOCAL_SITE_NAME, REMOTE_RESOURCE_URI, DEFAULT_METACARD_ID);
+        setupMockResourceCacheServiceMBeanProxy(false);
+        assertThat(actionProvider.canHandleMetacard(mockMetacard), is(true));
+    }
+
+    /**
+     * Metacard source id is local, Metacard contains content resource uri, resource is cached; no
+     * action returned
+     */
+    @Test
+    public void canHandleLocalSiteLocalUriResourceCached() throws Exception {
+        setupMockBasicMetacard(LOCAL_SITE_NAME, CONTENT_RESOURCE_URI, DEFAULT_METACARD_ID);
+        setupMockResourceCacheServiceMBeanProxy(true);
+        assertThat(actionProvider.canHandleMetacard(mockMetacard), is(false));
+    }
+
+    /**
+     * Metacard source id is local, Metacard contains content resource uri, resource is not cached;
+     * no action returned
+     */
+    @Test
+    public void canHandleLocalSiteLocalUriResourceNotCached() throws Exception {
+        setupMockBasicMetacard(LOCAL_SITE_NAME, CONTENT_RESOURCE_URI, DEFAULT_METACARD_ID);
+        setupMockResourceCacheServiceMBeanProxy(false);
+        assertThat(actionProvider.canHandleMetacard(mockMetacard), is(false));
+    }
+
+    /**
+     * Metacard source id is local, Metacard contains no resource uri; no action returned
+     */
+    @Test
+    public void canHandleLocalSiteNoUri() throws Exception {
+        setupMockBasicMetacard(LOCAL_SITE_NAME, null, DEFAULT_METACARD_ID);
+        setupMockResourceCacheServiceMBeanProxy(false);
+        assertThat(actionProvider.canHandleMetacard(mockMetacard), is(false));
+    }
+
+    /**
+     * Metacard source id is remote, Metacard contains no resource uri; no action returned
+     */
+    @Test
+    public void canHandleRemoteSiteNoUri() throws Exception {
+        setupMockBasicMetacard(REMOTE_SITE_NAME, null, DEFAULT_METACARD_ID);
+        setupMockResourceCacheServiceMBeanProxy(false);
+        assertThat(actionProvider.canHandleMetacard(mockMetacard), is(false));
+    }
```

Comment:
Parametrized tests (http://junit.org/junit4/javadoc/latest/org/junit/runners/Parameterized.html) would have been a good way to implement the above tests ;-)